setuptools/sandbox.py

```diff
@@ -411,7 +422,7 @@ def _remap_pair(self, operation, src, dst, *args, **kw):
 class DirectorySandbox(AbstractSandbox):
     """Restrict operations to a single subdirectory - pseudo-chroot"""
 
-    write_ops = dict.fromkeys([
+    write_ops: dict[str, None] = dict.fromkeys([
```

Comment:
The inferred type is `dict[str, Any | None]` to be exact (by both pyright and mypy). Because *we* know the value should never be used (this really ought to be a `set`, not a `dict`) we can be more precise on the type. Preventing some misuse like `self.write_ops["open"] = 1` and communicating more clearly that there's no use for the values of this dict.

The `| None` in `Any | None` did already prevent `test: str = self.write_ops["open"]`